client/app/vps/import/vps.service.js

```diff
@@ -276,10 +290,28 @@ angular.module('managerApp').service('VpsService', [
       return this.getSelectedVps(serviceName).then((vps) => {
         if (!ips) {
           return $q.reject('No ips');
-        } if (vps && vps.name) {
-          return $http.post([aapiRootPath, vps.name, 'ips', 'reverse'].join('/'), ips, { serviceType: 'aapi' })
-            .then((data) => { result = data.data; });
         }
+
+        if (vps && vps.name) {
+          const ip = _.head(ips.results);
+          const ipType = _.get(ip, 'type');
+
+          if (ipType === IP_PRIMARY_TYPE) {
+            return $http.post([aapiRootPath, vps.name, 'ips', 'reverse'].join('/'), ips, { serviceType: 'aapi' })
```

Comment:
Use string interpolation instead :)

``${aapiRootPath}/${vps.name}/ips/reverse``

Reply:
Not related to the fix but yes we can improve readability in a separate pull request.

Thanks for the suggestion.